src/org/zaproxy/zap/extension/ascanrulesAlpha/FormatString.java

```diff
@@ -0,0 +1,258 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A Format String  scan rule 
+ * Copyright (C) 2015 Institute for Defense Analyses
+ * @author Mark Rader based upon the example active scanner by psiinon
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE
+ */
+
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.core.scanner.AbstractAppParamPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.zap.model.Vulnerabilities;
+import org.zaproxy.zap.model.Vulnerability;
+import org.zaproxy.zap.network.HttpResponseBody;
+
+public class FormatString extends AbstractAppParamPlugin  {
+
+	// wasc_7 is a buffer overflow ;)
+	private static Vulnerability vuln = Vulnerabilities.getVulnerability("wasc_6");
+
+	private static Logger log = Logger.getLogger(FormatString.class);
+	
+	@Override
+	public int getId() {
+		/*
+		 * This should be unique across all active and passive rules.
+		 * The master list is http://code.google.com/p/zaproxy/source/browse/trunk/src/doc/alerts.xml
+		 */
+		return 30002;
+	}
+
+	@Override
+	public String getName() {
+		// Strip off the "Format String Error: " part if implementing a real one ;)
+		if (vuln != null) {
+			return "Format String Error " + vuln.getAlert();

```

Comment:
The concatenation should be removed, otherwise the name will be, in English, "Format String Error Format String".